assets/js/barChart.js

```diff
@@ -61,6 +61,10 @@ class BarChart {
         // Creates a d3 axis given a scale (takes care of tick marks and labels)
         let yAxis = d3.svg.axis()
             .scale(yScale)
+            // Limit label length to 50 characters plus ellipses.
+            .tickFormat(function(d) {
+                return d.substring(0, 50) + (d.length > 50 ? " ..." : "");
```

Comment:
Good point as always @jdaguil  .  I've created a static method: `static trimName(name) {}`; changed the variable `trimName` to `trimmedName`; and called the method like this: `trimmedName = BarChart.trimName(key);`.